src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py

```diff
@@ -16443,8 +16443,15 @@ def str_pad(
         width: int,
         side: Literal["left", "right", "both"] = "left",
         fillchar: str = " ",
-    ) -> None:
-        ErrorMessage.method_not_implemented_error("pad", "Series.str")
+    ) -> "SnowflakeQueryCompiler":
+        if side == "left":
+            return self.str_rjust(width, fillchar)
+        elif side == "right":
+            return self.str_ljust(width, fillchar)
+        elif side == "both":
+            return self.str_center(width, fillchar)
+        else:
+            raise ValueError("Invalid side")
```

Comment:
Nit: Can we have this ValueError mention the API?
```suggestion
            raise ValueError("Invalid side for Series.str.pad")
```

Reply:
This is mimicking the exact error message raised by native pandas! so I didn't want to change it.